plugins/fs/guest-js/index.ts

```diff
@@ -1075,9 +1075,18 @@ async function writeFile(
 
   if (data instanceof ReadableStream) {
     const file = await open(path, options)
-    for await (const chunk of data) {
-      await file.write(chunk)
+    const reader = data.getReader()
+  
+    try {
+      while (true) {
+        const { done, value } = await reader.read()
+        if (done) break
+        await file.write(value)
+      }
+    } finally {
+      reader.releaseLock()
     }
+  
     await file.close()
```

Comment:
I think we should propagate the errors if possible
```suggestion
    try {
      while (true) {
        const { done, value } = await reader.read()
        if (done) break
        await file.write(value)
      }
    } catch(e) {
      reader.releaseLock()
      await file.close()
      throw e
    }
```

Reply:
@amrbashir - with a `try / finally` block, errors will still propagate as expected, no? Also, I think we should release the lock and close the file handle, regardless of an error, so it seems like `finally` is the right place for this. Let me know your thoughts.

```js
function test() {
  try {
    throw new Error("Oops!");
  } finally {
    console.log("Finally block runs!");
  }
}

test(); // Logs "Finally block runs!", then throws "Oops!"
```

Reply:
this does work indeed, thank you